daemon/containerd/image_snapshot_unix.go

```diff
@@ -85,7 +95,50 @@ func (i *ImageService) unremapRootFS(ctx context.Context, mounts []mount.Mount)
 				return err
 			}
 
-			return os.Lchown(path, uid, gid)
+			return chownWithCaps(path, uid, gid)
 		})
 	})
 }
+
+// chownWithCaps will chown path and preserve the extended attributes.
+func chownWithCaps(path string, uid int, gid int) error {
+	xattrKeys, err := sysx.LListxattr(path)
+	if err != nil {
+		if !errors.Is(err, syscall.EOPNOTSUPP) {
+			return err
+		}
+	}
+
+	xattrs := make(map[string][]byte, len(xattrKeys))
+
+	for _, xattr := range xattrKeys {
+		data, err := sysx.LGetxattr(path, xattr)
```

Comment:
How does continuity's `sysx.LGetxattr()` differ from our `pkg/system.LGetxattr()` ? I haven't dug into it, but curious why one is used instead of the other.

Reply:
The two are the same, I'm using `sysx` here because it's the only one that has `LListxattr` so I thought I should use the same package for everything

Reply:
> I'm using sysx here because it's the only one that has LListxattr so I thought I should use the same package for everything

Ah! So I noticed that this is the _first_ use of this package in our code, and everything else uses `pkg/system`. We should look at that;

- consider if "continuity" is the right place for this (I know it's become a bit of a grab-bag of things)
- check what the origin of the "continuity" code is (was it forked from our `pkg/system` or vice-versa?)
- consider if we should move that functionality of `pkg/system` elsewhere (perhaps a module in https://github.com/moby/sys)
- in either case, consolidate those packages if they really overlap 90%, deprecate one (adding temporary aliases) and remove after.

From a quick look (see the `copyXattr` function I linked above), this appears to be because we only copy specific attributes;

https://github.com/moby/moby/blob/581cba2a567e95abeda9961329221694d0dcdcf1/daemon/graphdriver/copy/copy.go#L218
https://github.com/moby/moby/blob/581cba2a567e95abeda9961329221694d0dcdcf1/daemon/graphdriver/copy/copy.go#L275


:point_up: that leaves me the question; must we copy all attributes, or is this one of those cases where we should only copy specific ones?

For additional fun and giggles; https://github.com/tonistiigi/fsutil has a `copyXattrs` (which uses `sysx`) https://github.com/tonistiigi/fsutil/blob/797bd683d5a28a87bea5421f51db1e112ac2edd4/copy/copy_nowindows.go#L15C1-L32.